### PR TITLE
Make luigi package flake8 compliant

### PR DESCRIPTION
**Pull request by toidi**
`flake8` perform more source code checks then `pep8`.

Despite of there is some upcoming work, I made the step to improve code quality.

BTW, could you tell what is wrong with `doc/settings.py`? This file contain invalid python code.

### CODE REVIEW

File: test/server_test.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def __init__(self):
         self.unix_socket = os.path.join(self.temp_dir, 'luigid.sock')
 
     def run_server(self):
-        luigi.server.run(unix_socket=unix_socket)
+        luigi.server.run(unix_socket=self.unix_socket)
```
Comment (Tarrasch) [inline test/server_test.py]: !!!!!!!!!!!!!

@graingert, does this mean that tests haven't been running???

File: luigi/interface.py [hunk 2]
```diff
@@ -238,8 +238,6 @@ def parse_task(self, cmdline_args=None):
 
         _add_global_parameters(parser)
 
-        task_names = Register.task_names()
```
Comment (Tarrasch) [inline luigi/interface.py]: Did flake8 really find this??? How does it know the side effects are not needed?

File: luigi/__init__.py [hunk 3]
```diff
@@ -48,3 +48,14 @@
 from luigi.event import Event
 
 from .tools import range  # just makes the tool classes available from command line
+
+
+__all__ = [
```
Comment (Tarrasch) [inline luigi/__init__.py]: Hmm, what does this syntax mean?

**Comment by Tarrasch** [comment]
Looks really good! Thanks!

**Comment by Tarrasch** [comment]
Is flake as fast to run as pep8 by the way?

File: test/server_test.py (hunk 1 above)
Comment (toidi) [inline test/server_test.py]: I have no idea about that.

Original message was:
```
test/server_test.py:102:38: F821 undefined name 'unix_socket'
```

I just fixed that code. But if to look deeper - you can find that was dead code.

`UNIXServerClient` used by `UNIXServerTestRun` which in its turn have only `tearDown` method.

File: luigi/interface.py (hunk 2 above)
Comment (toidi) [inline luigi/interface.py]: Original message was:

```
luigi/interface.py:241:9: F841 local variable 'task_names' is assigned to but never used
```

`flake8` do only static code analysis. If `Register.task_names()` do something except returning task names, removing that line could be potentially dangerous.

File: luigi/__init__.py (hunk 3 above)
Comment (toidi) [inline luigi/__init__.py]: Documentation will describe better then me.

https://docs.python.org/2/tutorial/modules.html#importing-from-a-package

One the other side my motivation was remove warnings like:

```
luigi/__init__.py:34:1: F401 'Parameter' imported but unused
```

**Comment by toidi** [comment]
`flake8` utilize multi cpu and run faster then `pep8`.

Proof:

```
(luigi)luigi 🖖 time flake8 > /dev/null

real	0m1.157s
user	0m7.282s
sys	0m0.179s
(luigi)luigi 🖖 time pep8 > /dev/null

real	0m2.873s
user	0m2.851s
sys	0m0.019s
```

On the other side `flake8` plugins could make it more time consuming.

File: test/server_test.py (hunk 1 above)
Comment (graingert) [inline test/server_test.py]: Argh https://github.com/spotify/luigi/pull/1174/files#diff-9b595b54a300214f2172cde2e9c1eec4R179 is meant to inherit ServerTestRun!

Comment (Tarrasch) [inline test/server_test.py]: @graingert, can you please make sure things are properly tested? This is worrisome. :-/

**Comment by Tarrasch** [comment]
Ok, 7 seconds is fine I suppose.

I don't know about `doc/settings.py`, can it just be removed? :)

I think this can be merged once you get the build working and have squashed the commits.

File: tox.ini [hunk 4]
```diff
@@ -51,9 +51,11 @@ commands =
...
   coveralls
 
-[testenv:pep8]
-deps = pep8
-commands = pep8 --exclude=luigi/six.py --ignore E501,E402,E731 luigi test examples bin
+[testenv:flake8]
+deps = flake8
+commands = flake8 --exclude=doc,test,luigi/six.py --ignore=E501
+  flake8 --ignore=E265,E501 doc
+  flake8 --ignore=E501,F401,F841 test
```
Comment (graingert) [inline tox.ini]: Instead of ignoring E501 can you just set the line length limit to the longest line used?

Comment (toidi) [inline tox.ini]: Of course. What line length should be enough?

There are lines > 200 symbols length.

Here are some stats.

```
(luigi)luigi 🖖 flake8 --select=E501 --max-line-length=100 | wc -l
     512
(luigi)luigi 🖖 flake8 --select=E501 --max-line-length=120 | wc -l
     166
(luigi)luigi 🖖 flake8 --select=E501 --max-line-length=160 | wc -l
      28
(luigi)luigi 🖖 flake8 --select=E501 --max-line-length=200 | wc -l
       9
```

F401,F841 should also be fixed. Can we do it in separate pull request as well as line length?

Comment (graingert) [inline tox.ini]: What's the longest line?

Comment (toidi) [inline tox.ini]: 384

Comment (graingert) [inline tox.ini]: Set the limit to 385 and fix the rest in another PR?

Comment (toidi) [inline tox.ini]: Ok. What should be target line length?

Comment (graingert) [inline tox.ini]: Worry about that in another PR? We probably want to stick with the default

Comment (Tarrasch) [inline tox.ini]: Good decision guys!

**Comment by Tarrasch** [comment]
LGTM. Will merge when somebody else +1's.

**Comment by Tarrasch** [comment]
Actually two more small changes please:

 * Can you make the commit message properly capitalized, and you probably mean "compliant", not "complaint", right?. Something like "Make luigi package flake8 compliant"
 * Can you update all references to `pep8`? For example `CONTRIBUTING.rst`

**Comment by toidi** [comment]
Done

**Comment by Tarrasch** [comment]
Awesomeness!
